src/Aspire.Hosting/AppHostEnvironment.cs

```diff
@@ -0,0 +1,115 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Hosting;
+
+namespace Aspire.Hosting;
+
+/// <summary>
+/// Provides information about the AppHost environment.
+/// </summary>
+public sealed class AppHostEnvironment
+{
+    private readonly IConfiguration _configuration;
+    private readonly IHostEnvironment _hostEnvironment;
```

Comment:
@davidfowl - should this class exposes these fields as properties? That way consumers can get to this information directly, if necessary? And they don't need to inject multiple DI services.